packages/hardhat-core/src/internal/core/runtime-environment.ts

```diff
@@ -218,20 +223,23 @@ export class Environment implements HardhatRuntimeEnvironment {
   private async _runTaskDefinition(
     taskDefinition: TaskDefinition,
     taskArguments: TaskArguments,
+    subtaskArguments: SubtaskArguments,
     taskProfile?: TaskProfile
   ): Promise<any> {
     let runSuperFunction: any;
 
     if (taskDefinition instanceof OverriddenTaskDefinition) {
       runSuperFunction = async (
-        _taskArguments: TaskArguments = taskArguments
+        _taskArguments: TaskArguments = taskArguments,
+        _subtaskArguments: SubtaskArguments = subtaskArguments
       ) => {
         log("Running %s's super", taskDefinition.name);
 
         if (taskProfile === undefined) {
           return this._runTaskDefinition(
```

Comment:
There is another `this._runTaskDefinition` call after this that is not receiving the `_subtaskArguments`. I can't add a comment on that line, so I'm mentioning it here.